# HG changeset patch
# User Kartik Mohta <kartikmohta@gmail.com>
# Date 1496126419 14400
# Branch fix/GuiIface-strlen
# Node ID 2fac294590a187c851e95b4e3e7e34cdc5c76c6a
# Parent  bae84f76b4ad7a6a7cecaeb3e76e545b8c8d5f3e
GuiIface.cc: Fix buffer size for snprintf when putting "gazebo" into g_argv

diff --git a/gazebo/gui/GuiIface.cc b/gazebo/gui/GuiIface.cc
--- a/gazebo/gui/GuiIface.cc
+++ b/gazebo/gui/GuiIface.cc
@@ -303,8 +303,8 @@
   g_argv = new char*[g_argc];
   for (int i = 0; i < g_argc; i++)
   {
-    g_argv[i] = new char[strlen("gazebo")];
-    snprintf(g_argv[i], strlen("gazebo"), "gazebo");
+    g_argv[i] = new char[strlen("gazebo") + 1];
+    snprintf(g_argv[i], strlen("gazebo") + 1, "gazebo");
   }
 
   // Register custom message handler
